controllers/util/util.go

```diff
@@ -122,3 +126,77 @@ func GetSystemCerts() ([]byte, error) {
 	}
 	return data, err
 }
+
+func GetServiceHostname(ctx context.Context, svcName, ns string, client client.Client) (string, error) {
+	serviceHostname := ""
+	isAvailable, service, err := IsServiceHostnameAvailable(ctx, svcName, ns, client)
+	if err != nil {
+		return "", err
+	}
+
+	if isAvailable {
+		// Loop over all Service ports, if a secured port is found
+		// set port and scheme to its secured ones and exit the loop
+		servicePort := ""
+		serviceScheme := "http"
+		for port := range service.Spec.Ports {
+			servicePort = fmt.Sprintf("%d", port)
+			if servicePort == "8443" || servicePort == "443" {
+				// If a secured port is found, just set scheme to 'https://' and exit the loop
+				serviceScheme = "https"
+				break
+			}
+		}
+		serviceHostname = serviceScheme + "://" + service.Name + "." + service.Namespace + ".svc.cluster.local:" + servicePort
+	}
+	return serviceHostname, nil
+}
+
+func GetRouteHostname(ctx context.Context, routeName, ns string, client client.Client) (string, error) {
+	routeHostname := ""
+	isAvailable, route, err := IsRouteHostnameAvailable(ctx, routeName, ns, client)
+	if err != nil {
+		return "", err
+	}
+
+	if isAvailable {
+		serviceScheme := "http"
```

Comment:
`routeScheme` ... but I'd probably just call it `scheme`

same up in GetServiceHostname